cli/stream_command.go

```diff
@@ -1267,16 +1267,24 @@ func (c *streamCmd) copyAndEditStream(cfg api.StreamConfig, pc *fisk.ParseContex
 		cfg.Placement = &api.Placement{}
 	}
 
-	if cfg.Placement == nil {
-		cfg.Placement = &api.Placement{}
-	}
+	// We don't have a good way to signal "flag specified as explicitly empty" vs
+	// "flag not specified".  So we have to use sentinels.
+	// We use 'none' because - is dropped by the CLI lib before even reaching us.
```

Comment:
This whole stream edit model is deeply annoying and flawed needs a full redo so yeah, agree this is probably the least worse.  

What we can do is this:

```
f.Flag("discard-per-subject", "Sets the 'new' discard policy and applies it to every subject in the stream").IsSetByUser(&c.discardPerSubjSet).BoolVar(&c.discardPerSubj)
```

Here it would enable you to do `--placement-cluster=""` and you can determine the difference between default empty and specifically set empty using the `IsSetByUser()` variable.  In time as I get motivation I will probably add a  bunch of that for a hopefully less annoying outcome.

If we will treat `none` as magic we should call that out in the `--help` for this flag.

Reply:
I prefer the ability to sense an explicit empty string and not have magic sentinels which squat in the same namespace, documented or otherwise, so will rework this.